### DISCUSSION TITLE
Exclude filetypes/specific windows

### DISCUSSION THREAD

**Discussion by happenslol** [olimorris/persisted.nvim#117] [Q&A] [answered]
Hey, thanks for this great plugin! One issue that I have is that when I have a neo-tree window open when I close neovim, it will take up half of the screen (and be blank) when it's reopened, and I cannot close and reopen the window since it complains that the window has already been opened.

In general, I pretty much never want to save specific window types (plugin windows, quickfix list, etc.) to the session, only actually opened files. Is excluding file types in scope for this plugin?

**Comment by olimorris** [comment, marked as answer]
The only way of achieving this right now is by using [events](https://github.com/olimorris/persisted.nvim?tab=readme-ov-file#events--callbacks) and removing/closing the items before the session is saved (`PersistedSavePre`). You could run a command to check if NeoTree is open and execute a corresponding command.
